### PR TITLE
Frontend/handle unknown home state

### LINKED ISSUE #1006: Adjust Licensee Serializer to handle "unknown" in currentHomeJurisdiction json response
### Notes

- Currently, the Staff view of the licensingDetail page and the licensee list will have Unknown for home state and adressess for users who have not registered in the system and chosen a home state.
- We need to look specifically for "unknown" rather than falsey here and construct our existing fallbacks around this.
- Also add bestLicense() to run bestHomeStateLicense() and if nothing then for most recent active non homeStateLicense then most recent non home state license if the user has not selected a home state license
- Probably need to adjust the address serialization to only ever use the bestLicense's associated address as there  are no longer top level address fields returned in the licensee object

### Questions
1. Does falling back to `licenseJurisdiction` as a first step make sense if `currentHomeJurisdiction` is unknown? 

### Assumptions
1. Assumption text...

### Estimate: `?? hrs`
- Item 1: `?? hrs`
- TODO

### Tasks
- [ ] TODO
---
- [ ] Cross browser device testing
- [ ] Confirm `yarn build` is error & warning free
- [ ] Confirm `yarn serve` is error & warning free
- [ ] Confirm `yarn test:unit:all` is error & warning free
- [ ] Confirm `yarn test:e2e` is error & warning free
- [ ] QA Notes added
- [ ] PR opened with labels / reviewers / assignee / linked-issue

### QA Notes
- TODO

### PR DESCRIPTION
**Pull request by rmolinares**
### Requirements List
- _None_

### Description List
- Update Licensee model
    - Remove top-level address info (no longer returned by the API response)
    - Add ability to get "best" license with the priority order:
        1. Get the best license from the licensee's home state
        2. If no home jurisdiction license, look for active licenses in other states and select the most recently issued
        3. If no active non-home licenses exist, use the most recent inactive non-home license
        4. Return an empty License object if no licenses exist at all
    - Update home jurisdiction logic to use license jurisdiction state instead of current home jurisdiction if current home jurisdiction value coming from the API response is set to 'unknown'
- Update licensing detail page to display practitioner address from the newly refactored best license logic
- Update mock data to remove top-level address info

### Testing List
- `yarn test:unit:all` should run without errors or warnings
- `yarn serve` should run without errors or warnings
- `yarn build` should run without errors or warnings
- Code review
- Testing:
    - Log in as a compact admin and use the search licensing data list to comb through the list of practitioners and confirm that their address info is displayed as expected
        - Take a look at the API response in the Network tab if the browser and cross-compare with best license logic above   

Closes #1006 

## Summary by CodeRabbit

- **New Features**
  - Added a unified "best license" selector and a mailing-address accessor to surface the most relevant license and address.

- **Bug Fixes**
  - Improved home-jurisdiction fallback, tie-breaking, handling of missing/old issuance dates, inactive licenses, and address sourcing in the UI.

- **Chores**
  - Normalized fixture addresses/jurisdictions; removed deprecated home-jurisdiction address field; added licensee/provider identifiers; renamed issuance and download filename fields.

- **Documentation**
  - Updated compact-option references in public API docs.

- **Tests**
  - Expanded coverage for license selection, address resolution, cross-API consistency, and edge cases.

### CODE REVIEW

**Comment by rmolinares** [comment]
@jlkravitz This PR is now ready for review.

File: webroot/src/models/Licensee/Licensee.model.spec.ts [hunk 1]
```diff
@@ -668,19 +667,376 @@ describe('Licensee model', () => {
...
+                    issueDate: '2025-01-01',
+                    status: LicenseStatus.ACTIVE,
+                }),
+                new License({
+                    licenseNumber: 'co-same-date-2',
+                    issueState: new State({ abbrev: 'co' }),
+                    issueDate: '2025-01-01', // Same date
+                    status: LicenseStatus.ACTIVE,
+                })
+            ]
```
Comment (jlkravitz) [inline webroot/src/models/Licensee/Licensee.model.spec.ts:922]: Related to earlier comment, what if one of these is active and one is inactive? Not sure that was tested (e.g., more recent issue date, home state, but inactive).

File: webroot/src/models/Licensee/Licensee.model.spec.ts [hunk 2]
```diff
@@ -668,19 +667,376 @@ describe('Licensee model', () => {
...
+                    licenseeId: 'test-provider-id',
+                }),
+                // Inactive non-home license (Priority 3)
+                new License({
+                    licenseNumber: 'ma-inactive',
+                    issueState: new State({ abbrev: 'ma' }),
+                    issueDate: '2025-01-03',
+                    status: LicenseStatus.INACTIVE,
+                    licenseeId: 'test-provider-id',
+                })
```
[the remark this answers is not in the document]
Comment (rmolinares) [reply webroot/src/models/Licensee/Licensee.model.spec.ts]: Yeah. Good catch. I just added that scenario (priority 2) to the data and testing for that specifically here now.

**Review by jlkravitz — APPROVED** [review]
@isabeleliassen good to merge!
